test/common/CollectiveArgs.cpp

```diff
@@ -116,7 +110,7 @@ namespace RcclUnitTesting
   ErrCode CollectiveArgs::ValidateResults()
   {
     // Ignore non-root outputs for collectives with a root
-    if (CollectiveArgs::UsesRoot(this->funcType) && this->root != this->globalRank) return TEST_SUCCESS;
+    if (CollectiveArgs::UsesRoot(this->funcType) && this->options.root != this->globalRank) return TEST_SUCCESS;
```

Comment:
I think this might be broken then for Send/Recv.  
We should fix this by a function that tells us for each functionType/rank pair whether or not we need to check output.